docs/architecture/maui/accessing-remote-data.md

```diff
@@ -1,9 +1,380 @@
+
+## Making a POST request
+
+The `BasketService` class is used to manage the data retrieval and update process with the basket microservice. In the `RegisterAppServices` method in the `MauiProgram` class, the `BasketService` class is registered as a type mapping against the `IBasketService` type with the dependency injection container. Then, when an instance of the `BasketViewModel` class is created, its constructor accepts an `IBasketService` type, which the dependency injection container resolves, returning an instance of the `BasketService` class. For more information about dependency injection, see [Dependency Injection](dependency-injection.md).
+
+The image below shows the interaction of classes that send the basket data displayed by the BasketView, to the basket microservice.
+
+:::image type="content" source="media/sending-data-to-the-basket-microservice.png" alt-text="Sending data to the basket microservice.":::
+
+When an item is added to the shopping basket, the `ReCalculateTotalAsync` method in the `BasketViewModel` class is called. This method updates the total value of items in the basket, and sends the basket data to the basket microservice, as demonstrated in the following code example:
+
+```csharp
+private async Task ReCalculateTotalAsync()
+{
+
+    // Omitted for brevity...
+
+    await _basketService.UpdateBasketAsync(
+        new CustomerBasket
+        {
+            BuyerId = userInfo.UserId, 
+            Items = BasketItems.ToList()
+        }, 
+        authToken);
+}
+```
+
+This method calls the `UpdateBasketAsync` method of the `BasketService` instance that was injected into the `BasketViewModel` by the dependency injection container. The following method shows the `UpdateBasketAsync` method:
+
+```csharp
+public async Task<CustomerBasket> UpdateBasketAsync(CustomerBasket customerBasket, string token)
+{
+    UriBuilder builder = new UriBuilder(GlobalSetting.Instance.BasketEndpoint);
+    string uri = builder.ToString();
+    var result = await _requestProvider.PostAsync(uri, customerBasket, token);
+    return result;
+}
+```
+
+This method builds the URI that identifies the resource the request will be sent to, and uses the `RequestProvider` class to invoke the POST HTTP method on the resource, before returning the results to the `BasketViewModel`. Note that an access token, obtained from `IdentityServer` during the authentication process, is required to authorize requests to the basket microservice. For more information about authorization, see [Authorization](authentication-and-authorization.md).
+
+The following code example shows one of the `PostAsync` methods in the `RequestProvider` class:
+
+```csharp
+public async Task<TResult> PostAsync<TResult>(
+    string uri, TResult data, string token = "", string header = "")
+{
+    HttpClient httpClient = CreateHttpClient(token);
```

Comment:
Earlier you mentioned the importance of caching the `HttpClient` but here you seem to be creating a new one, no?

Reply:
This call will either create a new one or return a cached one, if available. I have the method shown above on line 95, but the name of this method could be updated to better reflect that. I could rename it to `GetHttpClient` or similar. Thoughts?